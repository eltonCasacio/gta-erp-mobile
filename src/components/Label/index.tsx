import React from 'react';
import {View, Text} from 'react-native';

const Label = () => {
  return (
    <>
      <View>
        <Text>Label</Text>
      </View>
    </>
  );
};

export default Label;
