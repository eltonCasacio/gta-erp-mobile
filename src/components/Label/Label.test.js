import React from 'react';
import renderer from 'react-test-renderer';

import Label from '.';

describe('<Label />', () => {
  test('Estrutura do layout', () => {
    const labelJson = renderer.create(<Label />).toJSON();
    const child = labelJson.children;

    expect(labelJson.type).toEqual('View');
    expect(child).toHaveLength(1);
    expect(child[0].type).toEqual('Text');

    expect(child[0].children).toHaveLength(1);
    expect(typeof child[0].children[0]).toEqual('string');

    expect(labelJson).toMatchSnapshot();
  });
});
