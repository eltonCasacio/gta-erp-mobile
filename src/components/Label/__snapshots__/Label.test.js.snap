// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Label /> Estrutura do layout 1`] = `
<View
  style={
    Array [
      Object {},
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#fff",
          "fontSize": 20,
        },
      ]
    }
  >
    label
  </Text>
</View>
`;
